### ISSUE TITLE
Multiple TelegramClient with dynamic creating.

### ISSUE THREAD

**Issue by osmancoskun** [LonamiWebs/Telethon#4037]
It would be nice to have a dynamically creating telegram client with Quart or Flask.
The application starts with getting the session stored on the same path and initializing it.
After that app run with Quart for listening to an IP address and doing some stuff. 
But when we create a telegram client with a telegram account and try to create another one with a totally different one, sessions are gone.
I couldn't find but if there is a document, example, or sample code about it, If you can share it, it would be fine. Otherwise, it would be a nice feature to have.

**Comment by osmancoskun** [comment]
With some logic and examples from github repositories, I ended up something like this. Currently I cant control session but I believe this can be improved.
```
import base64
import os
import glob
from quart import Quart, render_template_string, request

from telethon import TelegramClient, utils
from telethon.errors import SessionPasswordNeededError


def get_env(name, message):
    if name in os.environ:
        return os.environ[name]
    return input(message)


BASE_TEMPLATE = '''
<!DOCTYPE html>
<html>
    <head>
        <meta charset='UTF-8'>
        <title>Telethon + Quart</title>
    </head>
    <body>{{ content | safe }}</body>
</html>
'''

PHONE_FORM = '''
<form action='/' method='post'>
    Phone (international format): <input name='phone' type='text' placeholder='+34600000000'>
    <input type='submit'>
</form>
'''

CODE_FORM = '''
<form action='/' method='post'>
    Telegram code: <input name='code' type='text' placeholder='70707'>
    <input type='submit'>
</form>
'''

PASSWORD_FORM = '''
<form action='/' method='post'>
    Telegram password: <input name='password' type='text' placeholder='your password'>
    <input type='submit'>
</form>
'''

# Session name, API ID and hash to use; loaded from environmental variables
API_ID = 'XXXXXXXXXXXXXXXXX'
API_HASH = 'XXXXXXXXXXXXXXXXXXXXXXXX'

clients = {}
sessions = glob.glob("./*.sessions")
print(sessions)
if len(sessions) != 0:
    for ses in sessions:
        ses_phone = ses.split("./session")[0]
        tmp_session = os.environ.get("TG_SESSION",ses_phone)
        cli = TelegramClient(tmp_session,API_ID,API_HASH)
        clients[ses_phone] = cli


# Telethon client
#client = TelegramClient(SESSION, API_ID, API_HASH)
#client.parse_mode = 'html'  # <- Render things nicely
#phone = None

# Quart app
app = Quart(__name__)
app.secret_key = 'CHANGE THIS TO SOMETHING SECRET'


# Helper method to format messages nicely
async def format_message(message):
    if message.photo:
        content = '<img src="data:image/png;base64,{}" alt="{}" />'.format(
            base64.b64encode(await message.download_media(bytes)).decode(),
            message.raw_text
        )
    else:
        # client.parse_mode = 'html', so bold etc. will work!
        content = (message.text or '(action message)').replace('\n', '<br>')

    return '<p><strong>{}</strong>: {}<sub>{}</sub></p>'.format(
        utils.get_display_name(message.sender),
        content,
        message.date
    )


# Connect the client before we start serving with Quart
@app.before_serving
async def startup():
    # After connecting, the client will create additional asyncio tasks that run until it's disconnected again.
    # Be careful to not mix different asyncio loops during a client's lifetime, or things won't work properly!
    if len(clients) != 0:
        for phone in clients:
            await clients[phone].client.connect()
    


# After we're done serving (near shutdown), clean up the client
@app.after_serving
async def cleanup():
    if len(clients) != 0:
        for phone in clients:
            await clients[phone].client.disconnect()
    
    


@app.route('/', methods=['GET', 'POST'])
async def root():
    # We want to update the global phone variable to remember it
    # global phone
    # Check form parameters (phone/code)
    form = await request.form
    phone = form['phone']
    
    if 'phone' in form:
        phone = form['phone']
        clients[phone] = TelegramClient(phone,API_ID,API_HASH)
        await clients[phone].connect()
        await clients[phone].send_code_request(phone)

    if 'code' in form:
        try:
            await clients[phone].sign_in(code=form['code'])
        except SessionPasswordNeededError:
            return await render_template_string(BASE_TEMPLATE, content=PASSWORD_FORM)

    if 'password' in form:
        await clients[phone].sign_in(password=form['password'])

    # If we're logged in, show them some messages from their first dialog
    if await clients[phone].is_user_authorized():
        # They are logged in, show them some messages from their first dialog
        dialog = (await clients[phone].get_dialogs())[0]
        result = '<h1>{}</h1>'.format(dialog.title)
        async for m in clients[phone].iter_messages(dialog, 10):
            result += await(format_message(m))

        return await render_template_string(BASE_TEMPLATE, content=result)

    # Ask for the phone if we don't know it yet
    if phone is None:
        return await render_template_string(BASE_TEMPLATE, content=PHONE_FORM)

    # We have the phone, but we're not logged in, so ask for the code
    return await render_template_string(BASE_TEMPLATE, content=CODE_FORM)


# By default, `Quart.run` uses `asyncio.run()`, which creates a new asyncio
# event loop. If we had connected the `TelegramClient` before, `telethon` will
# use `asyncio.get_running_loop()` to create some additional tasks. If these
# loops are different, it won't work.
#
# To keep things simple, be sure to not create multiple asyncio loops!
if __name__ == '__main__':
    app.run()
```

**Comment by Lonami** [comment]
Telethon's "entry point" is a `TelegramClient`, which is a class. And Python lets the user create as many instances of a class as the user wishes (except for singletons, which I'm sure exist, but `TelegramClient` is not one of them).

This means what's being asked is already possible: you are free to create as many client instances as you want, dynamically or not. Telethon will have no trouble as long as you're not reusing the sessions.
